MerchiPlugin/PublicPages/ProductPage.php

```diff
@@ -138,92 +216,267 @@ public function custom_display_independent_attributes() {
+            
+            $html .= '<input type="checkbox" 
+                        id="' . $input_id . '"
+                        name="' . $name_prefix . '[' . $slug . '][]" 
+                        value="' . esc_attr($term->slug) . '"' . 
+                        $common_data_attrs . ' 
+                        data-variation-field-value="' . esc_attr($variation_option_id) . '"
+                        data-variation-unit-cost="' . esc_attr($variation_unit_cost) . '"
+                        class="image-select-input"/>';
+                        
+            $html .= '<label class="image-select-label" for="' . $input_id . '">';
+            $html .= '<div class="image-select-wrapper">';
+            if ($image_url) {
+                $html .= '<div class="image-container">';
+                $html .= '<img src="' . esc_url($image_url) . '" alt="' . esc_attr($term->name) . '" class="option-image">';
+                $html .= '<div class="image-select-checkmark"></div>';
+                $html .= '</div>';
+            }
+            $html .= '<span class="image-title">' . esc_html($term->name) . ' (+ $' . number_format($variation_unit_cost, 2) . ' per unit)</span>';
+            $html .= '</div>'; // Close image-select-wrapper
+            $html .= '</label>';
+            $html .= '</div>'; // Close image-select-option
+        }
+        $html .= '</div>';
+        $html .= '</fieldset>';
+    } 
+    // COLOUR_SELECT type (11)
+    else if ($field_type === 11) {
+        $html .= '<div class="color-options-grid">';
+        foreach ($terms as $index => $term) {
+            $is_checked = $index === 0 ? 'checked' : '';
+            $variation_option_id = get_term_meta($term->term_id, 'variation_option_id', true);
+            $variation_unit_cost = get_term_meta($term->term_id, 'variation_unit_cost', true);
+            $variation_unit_cost = is_numeric($variation_unit_cost) ? floatval($variation_unit_cost) : 0.0;
+            $color_value = strtolower($term->name);
+            $html .= '<label class="color-option">';
+            $html .= '<input type="radio" name="' . $name_prefix . '[' . $slug . ']" value="' . esc_attr($term->slug) . '" ' . $is_checked . $common_data_attrs . ' data-variation-field-value="' . esc_attr($variation_option_id) . '" data-variation-unit-cost="' . esc_attr($variation_unit_cost) . '"/>';
+            $html .= '<div class="color-option-inner">';
+            $html .= '<span class="color-indicator" style="background-color: ' . esc_attr($color_value) . ';"></span>';
+            $html .= '<span class="checkmark">✓</span>';
+            $html .= '</div>';
+            $html .= '<span class="color-name">' . esc_html($term->name) . ' (+ $' . number_format($variation_unit_cost, 2) . ' per unit)</span>';
+            $html .= '</label>';
+        }
+        $html .= '</div>';
+    }
 
-	$html .= '</div>';
-	return $html;
+    $html .= '</div>';
+    return $html;
 }
```

Comment:
I like this function and I think it should be the base for the structure of the function we build which will be responsible for rendering both the independentVariationFields and each group's groupVariationFields, however it starts with `// SELECT field type (2)` and then jumps to  `// CHECKBOX type (6)` and skips (8). We should define each filed explicitly for code readability.